pkg/rancher-components/types/index.d.ts

```diff
@@ -0,0 +1,11 @@
+import { VueConstructor } from 'vue'
+
+export const Badge: VueConstructor
```

Comment:
Typing all of our components a `VueConstructor` seems to give us basic typescript support. I'm wondering if we could do better here.. I know that this is improved in Vue 3

Reply:
There's an ongoing discussion to the methodology to be adopted for Typescript components and I'm updating the state in Confluence https://confluence.suse.com/pages/viewpage.action?pageId=969801913
We will have to discuss how to proceed for integrating TS in the components due the variety of cases.